packages/plexus/src/LayeredDirectedGraph/index.tsx

```diff
@@ -0,0 +1,220 @@
+    scaledStrokeWidth,
+  };
+
+  static defaultProps = {
+    className: '',
+    classNamePrefix: 'plexus',
+    minimap: false,
+    minimapClassName: '',
+    zoom: false,
+  };
+
+  state: TLayeredDirectedGraphState<T, U> = {
+    edges: [],
+    layoutEdges: null,
+    layoutGraph: null,
+    layoutPhase: ELayoutPhase.NoData,
+    layoutVertices: null,
+    sizeVertices: null,
+    vertices: [],
+    zoomTransform: zoomIdentity,
+  };
+
+  baseId = `plexus--LayeredDirectedGraph--${idCounter++}`;
+
+  rootRef: React.RefObject<HTMLDivElement> = React.createRef();
+
+  zoomManager: ZoomManager | null = null;
+
+  constructor(props: TLayeredDirectedGraphProps<T, U>) {
+    super(props);
+    const { edges, vertices, zoom: zoomEnabled } = props;
+    if (Array.isArray(edges) && edges.length && Array.isArray(vertices) && vertices.length) {
+      this.state.layoutPhase = ELayoutPhase.CalcSizes;
+      this.state.edges = edges;
+      this.state.vertices = vertices;
+    }
+    if (zoomEnabled) {
+      this.zoomManager = new ZoomManager(this.onZoomUpdated);
+    } else {
+      this.zoomManager = null;
+    }
+    this.renderUtils = {
+      getLocalId: this.getLocalId,
+      getZoomTransform: this.getZoomTransform,
+    };
+  }
+
+  componentDidMount() {
+    const { current } = this.rootRef;
+    if (current && this.zoomManager) {
+      this.zoomManager.setElement(current);
+    }
+  }
+
+  getLocalId = (name: string) => `${this.baseId}--${name}`;
+
+  getZoomTransform = () => this.state.zoomTransform;
+
+  private setSizeVertices = (sizeVertices: TSizeVertex<T>[]) => {
+    this.setState({ sizeVertices });
+    const { edges, layoutManager } = this.props;
+    const { layout } = layoutManager.getLayout(edges, sizeVertices);
+    // const { positions, layout } = layoutManager.getLayout(edges, sizeVertices);
+    // positions.then(this._onPositionsDone);
+    layout.then(this.onLayoutDone);
+    this.setState({ sizeVertices, layoutPhase: ELayoutPhase.CalcPositions });
+  };
+
+  private onZoomUpdated = (zoomTransform: ZoomTransform) => {
+    this.setState({ zoomTransform });
+  };
+
+  private onLayoutDone = (result: TCancelled | TLayoutDone) => {
+    if (result.isCancelled) {
+      return;
+    }
+    const { edges: layoutEdges, graph: layoutGraph, vertices: layoutVertices } = result;
+    this.setState({ layoutEdges, layoutGraph, layoutVertices, layoutPhase: ELayoutPhase.Done });
+    if (this.zoomManager) {
+      this.zoomManager.setContentSize(layoutGraph);
+    }
+  };
+
+  render() {
+    const {
+      className,
+      classNamePrefix,
+      edges,
+      minimap: minimapEnabled,
+      minimapClassName,
+      vertices,
+      setOnContainer,
+    } = this.props;
+    const { layoutEdges, layoutGraph, layoutVertices, zoomTransform } = this.state;
+    const children = React.Children.map(this.props.children, child => {
+      if (!isReactElement(child)) {
+        return child;
+      }
+      if (child.type !== HtmlScope) {
```

Comment:
Consider checking for `__DEV__` and rendering errors when compontsn are misused. E.g. measurebleNodesLayer is a direct child.